generic/tests/nfs_corrupt.py

```diff
@@ -215,7 +216,12 @@ def check_vm_status(vm, status):
         test.error("failed to create VM")
     session = vm.wait_for_login(timeout=int(params.get("login_timeout", 360)))
 
-    nfs_devname = get_nfs_devname(params, session)
+    if stg_params.get('blk_extra_params'):
+        pattern = re.search(r"(serial|wwn)=(\w+)",
+                            stg_params["blk_extra_params"], re.M)
+        if pattern is not None:
```

Comment:
With this change, no need to check the pattern. The only concern is that if users re-define the blk_extra_params, then we can not make sure serial=xxx exists. But if no nfs_devname, this case will fail.

Reply:
We can consider it is pre-condition.  otherwise if serial non-existent, the other keys also may non-existent.  They are not the test points.
If the expected key does not exist result in failed result, the coder should check its code or configuration.   

Anyway the both are accepable to me.

Reply:
updated, please check if the new update could fix the issue